### PR TITLE
Add default config for cnn policies

### PR DESCRIPTION
**Pull request by anshradh**
## Description

Adds a default configuration option for CNN policies, addressing #549 

## Testing

Tested with following command: 
`python -m imitation.scripts.train_rl with common.env_name=AsteroidsNoFrameskip-v4 train.cnn_policy`

### CODE REVIEW

File: src/imitation/scripts/config/train_rl.py [hunk 1]
```diff
@@ -44,6 +46,16 @@ def default_end_cond(rollout_save_n_timesteps, rollout_save_n_episodes):
         rollout_save_n_timesteps = 2000  # Min timesteps saved per file, optional.
 
 
+# Default config for CNN Policies
+@train_rl_ex.named_config
+def cnn_policy():
+    train = dict(
+        policy_cls=CnnPolicy,
+        policy_kwargs=dict(features_extractor_class=NatureCNN),
```
Comment (AdamGleave) [inline src/imitation/scripts/config/train_rl.py:54]: We should probably remove the config hook in `common/train.py` and then remove `policy_kwargs`.

Should probably also move the named config itself into `common/train.py`.

Comment (anshradh) [reply src/imitation/scripts/config/train_rl.py]: Do you have pointers to where I should update documentation since I'm removing the hook?

File: tests/scripts/test_scripts.py [hunk 2]
```diff
@@ -723,6 +723,29 @@ def test_train_rl_double_normalization(tmpdir: str, rng):
...
 
 
+def test_train_rl_cnn_policy(tmpdir: str, rng):
+    venv = util.make_vec_env(
+        "AsteroidsNoFrameskip-v4",
+        n_envs=1,
+        parallel=False,
+        rng=rng,
+    )
+    print(venv)
```
Comment (AdamGleave) [inline tests/scripts/test_scripts.py:733]: Do you want to leave this in?

File: tests/scripts/test_scripts.py [hunk 3]
```diff
@@ -723,6 +723,29 @@ def test_train_rl_double_normalization(tmpdir: str, rng):
...
+    net = reward_nets.CnnRewardNet(venv.observation_space, venv.action_space)
+    tmppath = os.path.join(tmpdir, "reward.pt")
+    th.save(net, tmppath)
+
+    log_dir_data = os.path.join(tmpdir, "train_rl")
+    with pytest.warns(RuntimeWarning):
```
Comment (AdamGleave) [inline tests/scripts/test_scripts.py:739]: What warning are we expecting to see here? We don't have any warning specifically for this case do we?

(Separately I'm worried the other tests checking for `RuntimeWarning` are too vague, they might be passing because of other warnings that always get raised, we should probably add a match to check for the relevant message.)

File: tests/scripts/test_scripts.py [hunk 4]
```diff
@@ -723,6 +723,29 @@ def test_train_rl_double_normalization(tmpdir: str, rng):
...
+        train_rl.train_rl_ex.run(
+            named_configs=["train.cnn_policy"] + ALGO_FAST_CONFIGS["rl"],
+            config_updates=dict(
+                common=dict(log_dir=log_dir_data, env_name="AsteroidsNoFrameskip-v4"),
+                reward_path=tmppath,
+            ),
+        )
+
```
Comment (AdamGleave) [inline tests/scripts/test_scripts.py:747]: Consider checking:
```
    assert run.status == "COMPLETED"
    assert isinstance(run.result, dict)
```

like in `test_train_rl_main`.

Right now I think this test could pass even if there were some exceptions as Sacred will just capture them (but return `run.status == "ERROR"` or the like.)

File: src/imitation/scripts/config/train_rl.py (hunk 1 above)
Comment (AdamGleave) [reply src/imitation/scripts/config/train_rl.py]: On reflection I think we basically don't have the script configs documented, which is sad :( So maybe it's OK to not update the docs. Pro tip: the documentation can't be out of date if you don't have it...

Should update the comment in `normalize_disable` at least though as that's now out of date (mentions the now-removed config hook).

**Review by AdamGleave — CHANGES_REQUESTED** [review]
Config itself looks good. Some questions about the test case.

File: tests/scripts/test_scripts.py (hunk 3 above)
Comment (anshradh) [reply tests/scripts/test_scripts.py:739]: Yeah good point - removed checking for a warning here (doesn't really make sense). Should I make an issue for checking for the right messages on some of the other tests?

Comment (AdamGleave) [reply tests/scripts/test_scripts.py]: Yes, please make an issue. If you could also make a PR to solve that issue that'd be great, but it's definitely out of scope for this PR.

**Comment by AdamGleave** [comment]
> I think we can actually just remove the `normalize_disable` named_config? Seems like it's not needed after removing the config hook

Yeah, that seems right. There might be some edge case where we want it around (some policy with weird default?), but it's not hard to add back in later if we need it, so may as well simplify config code and remove it.

File: tests/scripts/test_scripts.py (hunk 4 above)
Comment (anshradh) [reply tests/scripts/test_scripts.py:747]: Done.

File: tests/scripts/test_scripts.py (hunk 2 above)
Comment (anshradh) [reply tests/scripts/test_scripts.py]: Haha nope, thanks.

File: tests/scripts/test_scripts.py (hunk 3 above)
Comment (anshradh) [reply tests/scripts/test_scripts.py]: Looks like there's actually just one test that does this, made an issue at #614. Will put up a PR in a bit to address.

**Review by AdamGleave — APPROVED** [review]
LGTM
